.github/workflows/ci.yml

```diff
@@ -12,90 +12,125 @@ on:
-      
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
       redis:
-        image: redis:alpine
-        ports: 
-          - 6379/tcp
+        image: public.ecr.aws/docker/library/redis:7-alpine
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
 
     steps:
-      - uses: actions/checkout@v4
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
-        with:
-          python-version: ${{ matrix.python-version }}
-      - name: Install Dependencies
-        run: |
-          python -m pip install --upgrade pip
-          python -m pip install tox~=4.4.11
+      - name: Checkout
+        uses: actions/checkout@v4
+
       - name: Build communication container
         working-directory: ./resolwe/flow/docker_images
-        run: docker build -f Dockerfile.communication -t resolwe/com:testing ../../
-      - name: Run Tests
-        run: >
-          RESOLWE_COMMUNICATOR_IMAGE=resolwe/com:testing
-          tox -e ${{ matrix.tox-env }}
-          --skip-missing-interpreters false
-          --recreate
-        env:
-          RESOLWE_REDIS_PORT: ${{ job.services.redis.ports[6379] }}
-          RESOLWE_POSTGRESQL_HOST: localhost
-          RESOLWE_POSTGRESQL_NAME: resolwe
-          RESOLWE_POSTGRESQL_USER: resolwe
-          RESOLWE_POSTGRESQL_PASS: resolwe
-          RESOLWE_POSTGRESQL_PORT: ${{ job.services.postgres.ports[5432] }}
-      - name: Upload coverage to Codecov
-        if: matrix.tox-env == 'py310' || matrix.tox-env == 'py311'
-        uses: codecov/codecov-action@v4
-        with:
-          file: .coverage
-      
-  build:
-    runs-on: ubuntu-22.04
+        run: docker build -f Dockerfile.communication -t $RESOLWE_COMMUNICATOR_IMAGE ../../
+
+      - name: Test
+        run: |
+          # This is the same directory as before, but the path must match
+          # the one on the host, so new docker containers can mount files in it.
+          cd /home/runner/_work/resolwe/resolwe
+          # Make sure `python` refers to the tox environment's Python
+          export PATH=`pwd`/.tox/${{ matrix.toxenv }}/bin:$PATH
+          # job.container.network is uninitialized at the time of setting environment variables
+          RESOLWE_DOCKER_NETWORK=${{ job.container.network }} ${{ env.TOX_COMMAND }} -e ${{ matrix.toxenv }}
 
+  build:
+    runs-on: arc-runner
     needs: test
-    
-    if: "startsWith(github.ref, 'refs/tags')"
+    if: startsWith(github.ref, 'refs/tags')
 
     steps:
       - uses: actions/checkout@v4
-      - name: Set up Python 3.11
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.11"
+
       - name: Install build
-        run: python -m pip install --user build
+        run: python3.12 -m pip install --user build
```

Comment:
Are you sure it is OK to use the Python version that was not used in tests to create a build?